website/content/docs/architecture/improving-consul-resilience.mdx

```diff
@@ -132,3 +123,25 @@ For more information on redundancy zones, refer to:
   for a more detailed explanation
 - [Redundancy zone tutorial](/consul/tutorials/enterprise/redundancy-zones)
   to learn how to use them
+
+### Autopilot
+
+Autopilot is a set of functionalities that include cleanup of dead servers, monitoring the state of the Raft protocol in the Consul cluster, and stable server introduction. Autopilot is enabled by default since Consul version `1.0`.
+
+With Autopilot's dead server cleanup enabled, failed servers will be marked as `Left` and removed from the Raft peer set to prevent them from interfering with the quorum size. Autopilot does that as soon as a replacement Consul server comes online. This is beneficial when server nodes have failed and have been redeployed, but their IP address and hostnames are no longer the same, and Consul considers them as new nodes. This functionality keeps the cluster peer set size correct and the quorum requirement simple.
+
+Take into consideration the following situation with Autopilot enabled: there is a cluster of 5 Consul server nodes. Therefore, quorum is 3 and the cluster can afford to lose only two nodes before the cluster fails. Two of the server nodes fail. These two nodes are redeployed in a cloud environment, which assigns new hostnames and IP addresses to the replacement nodes. Once they rejoin the cluster, Consul treats them as two extra nodes, unrelated to the previously failed. The cluster has now 3 surviving nodes, 2 failed nodes and 2 newly joined nodes - 7 in total. However, because Autopilot is enabled and two new nodes have been introduced to the cluster, it cleans up the entries for the two `Failed` nodes. The final state of the cluster is 3 surviving nodes and 2 newly joined nodes - 5 nodes in total. Quorum is still 3, so the cluster can afford to lose two nodes, and level of redundancy is preserved instantaneously.
+
+Take into consideration the following situation without Autopilot disabledenabled: there is a cluster of 5 Consul server nodes. Therefore, quorum is 3 and the cluster can afford to lose only two nodes before the cluster fails. Two of the server nodes fail. These two nodes are redeployed in a cloud environment, which assigns new hostnames and IP addresses to the replacement nodes. Once they rejoin the cluster, Consul treats them as two extra nodes, unrelated to the previously failed. The cluster has now 3 surviving nodes, 2 failed nodes and 2 newly joined nodes - 7 in total. However, because Autopilot is not enabled, introducing two new nodes to the cluster does not trigger the clean-up for the two `Failed` nodes. The final state of the cluster is 3 surviving nodes, 2 failed, and 2 newly joined nodes - 7 nodes in total. Quorum is now 4, so the cluster can afford to lose only one node until the two failed nodes are reaped in 72 hours. Therefore, the level of redundancy has decreased compared to the initial state.

```

Comment:
```suggestion
Take into consideration the following situation without Autopilot enabled: there is a cluster of 5 Consul server nodes. Therefore, quorum is 3 and the cluster can afford to lose only two nodes before the cluster fails. Two of the server nodes fail. These two nodes are redeployed in a cloud environment, which assigns new hostnames and IP addresses to the replacement nodes. Once they rejoin the cluster, Consul treats them as two extra nodes, unrelated to the previously failed. The cluster has now 3 surviving nodes, 2 failed nodes and 2 newly joined nodes - 7 in total. However, because Autopilot is not enabled, introducing two new nodes to the cluster does not trigger the clean-up for the two `Failed` nodes. The final state of the cluster is 3 surviving nodes, 2 failed, and 2 newly joined nodes - 7 nodes in total. Quorum is now 4, so the cluster can afford to lose only one node until the two failed nodes are reaped in 72 hours. Therefore, the level of redundancy has decreased compared to the initial state.

```